src/CP/valueselection/learning/rewards/smartreward.jl

```diff
@@ -19,15 +23,19 @@ function set_reward!(::Type{StepPhase}, lh::LearnedHeuristic{SR, SmartReward, A}
     A <: ActionOutput
 }
     if symbol == :Infeasible  
-        #println("INFEASIBLE")
-        #lh.reward.value -= last_episode_total_reward(lh.agent.trajectory)
         lh.reward.value -= 0
-
-    elseif symbol == :FoundSolution
-        #println("SOLUTION FOUND, score : ",assignedValue(model.objective), " delta : ",15-assignedValue(model.objective)," accumulated reward :  ", model.statistics.AccumulatedRewardBeforeReset)
-        lh.reward.value += isnothing(model.objective) ?  0 : 100 * (-assignedValue(model.objective))
-        #lh.reward.value += model.statistics.lastPruning
-
+    elseif symbol == :FoundSolution #last portion required to get the full closed path
+        dist = model.adhocInfo[1]
+        n =  size(dist)[1]
+        max_dist = Float32(Base.maximum(dist))
+            if isbound(model.variables["a_"*string(n-1)])
+                last = assignedValue(model.variables["a_"*string(n-1)])
+                first = assignedValue(model.variables["a_1"])
+        
+                dist_to_first_node = lh.current_state.dist[last, first] * max_dist
+                print("final_dist : ", dist_to_first_node, " // ")
+                lh.reward.value += -ρ*dist_to_first_node 
+            end
```

Comment:
This corresponds to the last path portion between the last assigned variable and the start variable.